stl/inc/type_traits

```diff
@@ -86,29 +87,46 @@ struct _Arg_types<_Ty1, _Ty2> {
     _CXX17_DEPRECATE_ADAPTOR_TYPEDEFS typedef _Ty1 first_argument_type;
     _CXX17_DEPRECATE_ADAPTOR_TYPEDEFS typedef _Ty2 second_argument_type;
 };
+#endif // !_HAS_CXX20
 
 // STRUCT TEMPLATE is_function
 template <class _Ty>
 struct _Is_function { // determine whether _Ty is a function
     using _Bool_type = false_type; // NB: members are user-visible via _Weak_types
 };
 
+#if !_HAS_CXX20
 #define _IS_FUNCTION(CALL_OPT, CV_OPT, REF_OPT, NOEXCEPT_OPT)                                           \
     template <class _Ret, class... _Types>                                                              \
     struct _Is_function<_Ret CALL_OPT(_Types...) CV_OPT REF_OPT NOEXCEPT_OPT> : _Arg_types<_Types...> { \
         using _Bool_type = true_type;                                                                   \
         _CXX17_DEPRECATE_ADAPTOR_TYPEDEFS typedef _Ret result_type;                                     \
     };
+#else // ^^^ !_HAS_CXX20 / _HAS_CXX20 vvv
+#define _IS_FUNCTION(CALL_OPT, CV_OPT, REF_OPT, NOEXCEPT_OPT)                                           \
+    template <class _Ret, class... _Types>                                                              \
+    struct _Is_function<_Ret CALL_OPT(_Types...) CV_OPT REF_OPT NOEXCEPT_OPT> : _Arg_types<_Types...> { \
+        using _Bool_type = true_type;                                                                   \
+    };
+#endif // _HAS_CXX20
 
 _NON_MEMBER_CALL_CV_REF_NOEXCEPT(_IS_FUNCTION)
 #undef _IS_FUNCTION
 
+#if !_HAS_CXX20
 #define _IS_FUNCTION_ELLIPSIS(CV_REF_NOEXCEPT_OPT)                                                            \
     template <class _Ret, class... _Types>                                                                    \
     struct _Is_function<_Ret(_Types..., ...) CV_REF_NOEXCEPT_OPT> { /* no calling conventions for ellipsis */ \
         using _Bool_type = true_type;                                                                         \
         _CXX17_DEPRECATE_ADAPTOR_TYPEDEFS typedef _Ret result_type;                                           \

```

Comment:
Instead of needing to duplicate this macro spam I think you could just make the typedef have an `_Ugly` name in 20 like this:

```
#if _HAS_CXX20
#define _RESULT_TYPE_NAME _Unnamable
#else
#define _RESULT_TYPE_NAME result_type
#endif // _HAS_CXX20
```

then name the typedef with that.